## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-xl.sh | :x: [more...](#c7dbda91-89b4-46d3-8f10-200350931661) |
| mint-large-bucket.sh | :x: [more...](#a247b9a1-46c2-4222-8130-9cdbd9a06aaa) |
| mint-dist-xl.sh | :x: [more...](#af24cf7c-8ab5-4f15-bc19-82aa71f065aa) |
<a name="a247b9a1-46c2-4222-8130-9cdbd9a06aaa"></a>
### 9902-5e62a2b/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev4.minio.io:32159
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 0805ad51b44b:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... FAILED in 4 minutes and 5 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-e3xqxt0jvj5gba",
    "expiry": 60000000000,
    "objectName1": "testObject1",
    "objectName2": "testObject2"
  },
  "duration": 120558,
  "error": "BucketAlreadyOwnedByYou: Your previous request to create the named bucket succeeded and you already own it.\n\tstatus code: 409, request id: 161B77E47AAED176, host id: ",
  "function": "testListObjects",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(1/15) Running aws-sdk-java tests ... done in 8 seconds
(2/15) Running aws-sdk-php tests ... FAILED in 18 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-e3xqxt0jvj5gba",
    "expiry": 60000000000,
    "objectName1": "testObject1",
    "objectName2": "testObject2"
  },
  "duration": 120558,
  "error": "BucketAlreadyOwnedByYou: Your previous request to create the named bucket succeeded and you already own it.\n\tstatus code: 409, request id: 161B77E47AAED176, host id: ",
  "function": "testListObjects",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(2/15) Running aws-sdk-ruby tests ... FAILED in 10 seconds
{
  "name": "aws-sdk-ruby",
  "function": "makeBucket(bucket_name)",
  "args": {
    "bucket_name": "aws-sdk-ruby-bucket-c249aca81634"
  },
  "duration": 343.05,
  "error": "Your previous request to create the named bucket succeeded and you already own it.",
  "status": "FAIL"
}
(2/15) Running awscli tests ... FAILED in 23 seconds
{
  "name": "awscli",
  "duration": 15665,
  "function": "delete_bucket\n",
  "status": "FAIL",
  "error": "delete: s3://awscli-mint-test-bucket-11080/datafile-1-kBremove_bucket failed: s3://awscli-mint-test-bucket-11080 An error occurred (BucketNotEmpty) when calling the DeleteBucket operation: The bucket you tried to delete is not empty"
}
(2/15) Running healthcheck tests ... done in 0 seconds
(3/15) Running mc tests ... FAILED in 5 minutes and 19 seconds
{
  "name": "mc",
  "duration": "88638",
  "function": "test_mirror_list_objects_storage_class",
  "status": "FAIL",
  "error": "/mint/run/core/mc/mc --config-dir /tmp/.mc-14772 --quiet --no-color mirror --storage-class REDUCED_REDUNDANCY /mint/data myminio/mc-test-bucket-20021  >>> \n`/mint/data/datafile-0-b` -> `myminio/mc-test-bucket-20021/datafile-0-b`\n`/mint/data/datafile-1-MB` -> `myminio/mc-test-bucket-20021/datafile-1-MB`\n`/mint/data/datafile-1-kB` -> `myminio/mc-test-bucket-20021/datafile-1-kB`\n`/mint/data/datafile-1-b` -> `myminio/mc-test-bucket-20021/datafile-1-b`\n`/mint/data/datafile-10-MB` -> `myminio/mc-test-bucket-20021/datafile-10-MB`\n`/mint/data/datafile-1.03-MB` -> `myminio/mc-test-bucket-20021/datafile-1.03-MB`\n`/mint/data/datafile-100-kB` -> `myminio/mc-test-bucket-20021/datafile-100-kB`\n`/mint/data/datafile-10-kB` -> `myminio/mc-test-bucket-20021/datafile-10-kB`\n`/mint/data/datafile-129-MB` -> `myminio/mc-test-bucket-20021/datafile-129-MB`\n`/mint/data/datafile-11-MB` -> `myminio/mc-test-bucket-20021/datafile-11-MB`\n`/mint/data/datafile-5-MB` -> `myminio/mc-test-bucket-20021/datafile-5-MB`\n`/mint/data/datafile-33-kB` -> `myminio/mc-test-bucket-20021/datafile-33-kB`\n`/mint/data/datafile-6-MB` -> `myminio/mc-test-bucket-20021/datafile-6-MB`\n`/mint/data/datafile-65-MB` -> `myminio/mc-test-bucket-20021/datafile-65-MB`\n`/mint/data/datafile-5243880-b` -> `myminio/mc-test-bucket-20021/datafile-5243880-b`\nmc: <ERROR> Failed to copy `/mint/data/datafile-129-MB`. One or more of the specified parts could not be found.  The part may not have been uploaded, or the specified entity tag may not match the part's entity tag.\nmc: <ERROR> Failed to copy `/mint/data/datafile-65-MB`. Please reduce your request\nmc: <ERROR> Failed to copy `/mint/data/datafile-1-b`. Please reduce your request\nmc: <ERROR> Failed to copy `/mint/data/datafile-5-MB`. Please reduce your request\nTotal: 0 B, Transferred: 998.23 MiB, Speed: 11.62 MiB/s"
}
(3/15) Running minio-dotnet tests ... done in 3 minutes and 24 seconds
(4/15) Running minio-go tests ... FAILED in 4 seconds
{
  "args": {
    "bucketName": "minio-go-test-4qzj2uhcyglk6tfk",
    "objectName": "paalrshs61cdixhocwwtdumrtuveqc",
    "region": "eu-west-1"
  },
  "duration": 4071,
  "error": "The bucket you tried to delete is not empty",
  "function": "MakeBucket(bucketName, region)",
  "message": "Cleanup failed",
  "name": "minio-go: testGetObjectClosedTwiceV2",
  "status": "FAIL"
}
(4/15) Running minio-java tests ... FAILED in 4 seconds
{
  "name": "minio-java",
  "function": "listBuckets()",
  "duration": 32,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = SlowDown, message = Please reduce your request, bucketName = minio-java-test-2gpunlu, objectName = null, resource = /minio-java-test-2gpunlu, requestId = 161B786E55D75D80, hostId = d3d4b19b-c669-49ae-abd1-27362afcf836)\nrequest={method=PUT, url=http://minio-dev4.minio.io:32159/minio-java-test-2gpunlu, headers=Host: minio-dev4.minio.io:32159\nAccept-Encoding: identity\nUser-Agent: MinIO (amd64; amd64) minio-java/dev\nContent-MD5: 1B2M2Y8AsgTpgAmY7PhCfg==\nx-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\nx-amz-date: 20200624T114828Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20200624/us-east-1/s3/aws4_request, SignedHeaders=accept-encoding;content-md5;host;x-amz-content-sha256;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=503, headers=Accept-Ranges: bytes\nContent-Length: 305\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nRetry-After: 120\nServer: MinIO/DEVELOPMENT.2020-06-24T11-28-44Z\nVary: Origin\nX-Amz-Request-Id: 161B786E55D75D80\nX-Xss-Protection: 1; mode=block\nDate: Wed, 24 Jun 2020 11:48:28 GMT\n}\n >>> [io.minio.MinioClient.execute(MinioClient.java:1211), io.minio.MinioClient.execute(MinioClient.java:1236), io.minio.MinioClient.executePut(MinioClient.java:1410), io.minio.MinioClient.makeBucket(MinioClient.java:3286), io.minio.MinioClient.makeBucket(MinioClient.java:3199), FunctionalTest.listBuckets_test(FunctionalTest.java:324), FunctionalTest.runTests(FunctionalTest.java:4011), FunctionalTest.main(FunctionalTest.java:4192)]"
}
(4/15) Running minio-js tests ... FAILED in 13 seconds
{
  "name": "minio-js",
  "function": "\"after all\" hook in \"functional tests\"",
  "duration": 36,
  "status": "FAIL",
  "error": "S3Error: Please reduce your request at Object.parseError (node_modules/minio/dist/main/xml-parsers.js:86:11) at /mint/run/core/minio-js/node_modules/minio/dist/main/transformers.js:156:22 at DestroyableTransform._flush (node_modules/minio/dist/main/transformers.js:80:10) at DestroyableTransform.prefinish (node_modules/readable-stream/lib/_stream_transform.js:129:10) at prefinish (node_modules/readable-stream/lib/_stream_writable.js:611:14) at finishMaybe (node_modules/readable-stream/lib/_stream_writable.js:620:5) at endWritable (node_modules/readable-stream/lib/_stream_writable.js:643:3) at DestroyableTransform.Writable.end (node_modules/readable-stream/lib/_stream_writable.js:571:22) at IncomingMessage.onend (_stream_readable.js:673:10) at endReadableNT (_stream_readable.js:1201:12) at processTicksAndRejections (internal/process/task_queues.js:84:21)"
}
(4/15) Running minio-py tests ... FAILED in 2 minutes and 2 seconds
{
  "name": "minio-py:test_list_buckets",
  "function": "list_buckets()",
  "duration": 120066,
  "message": "BucketAlreadyOwnedByYou: message: Your previous request to create the named bucket succeeded and you already own it.",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 270, in test_list_buckets\n    client.make_bucket(bucket_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 374, in make_bucket\n    raise ResponseError(response, method, bucket_name).get_exception()\nminio.error.BucketAlreadyOwnedByYou: BucketAlreadyOwnedByYou: message: Your previous request to create the named bucket succeeded and you already own it.\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 1979, in main\n    test_list_buckets(client, log_output)\n  File \"/mint/run/core/minio-py/tests.py\", line 279, in test_list_buckets\n    raise Exception(err)\nException: BucketAlreadyOwnedByYou: message: Your previous request to create the named bucket succeeded and you already own it.\n",
  "status": "FAIL"
}
(4/15) Running s3cmd tests ... done in 36 seconds
(5/15) Running s3select tests ... FAILED in 26 minutes and 20 seconds
{
  "name": "s3select:test_csv_input_quote_char",
  "function": "select_object_content(bucket_name, object_name, opts)",
  "args": {
    "bucket_name": "s3select-test-054ffeed-81c5-4c8b-a63e-43498f0b507c"
  },
  "duration": 1579403,
  "message": "BucketNotEmpty: message: The bucket you tried to delete is not empty.",
  "error": "Traceback (most recent call last):\n  File \"./tests.py\", line 45, in main\n    test_csv_input_custom_quote_char(client, log_output)\n  File \"/mint/run/core/s3select/csv.py\", line 110, in test_csv_input_custom_quote_char\n    client.remove_bucket(bucket_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 446, in remove_bucket\n    self._url_open('DELETE', bucket_name=bucket_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 2020, in _url_open\n    object_name).get_exception()\nminio.error.BucketNotEmpty: BucketNotEmpty: message: The bucket you tried to delete is not empty.\n",
  "status": "FAIL"
}
(5/15) Running security tests ... done in 0 seconds

Executed 5 out of 15 tests successfully.
```
<a name="af24cf7c-8ab5-4f15-bc19-82aa71f065aa"></a>
### 9902-5e62a2b/mint-dist-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev7.minio.io:32037
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 302869b6965a:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... FAILED in 6 minutes and 13 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-zgbchko5o1slbu",
    "expiry": 60000000000,
    "objectName1": "testObject1",
    "objectName2": "testObject2"
  },
  "duration": 249145,
  "error": "BucketNotEmpty: The bucket you tried to delete is not empty\n\tstatus code: 409, request id: 161B7802647B6105, host id: ",
  "function": "testListObjects",
  "message": "AWS SDK Go DeleteBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(1/15) Running aws-sdk-java tests ... done in 3 seconds
(2/15) Running aws-sdk-php tests ... done in 1 minutes and 1 seconds
(3/15) Running aws-sdk-ruby tests ... FAILED in 7 seconds
{
  "name": "aws-sdk-ruby",
  "function": "removeBucket(bucket_name)",
  "args": {
    "bucket_name": "aws-sdk-ruby-bucket-957964c9bd94"
  },
  "duration": 3511.92,
  "error": "Failed to clean-up bucket 'aws-sdk-ruby-bucket-957964c9bd94', The bucket you tried to delete is not empty",
  "status": "FAIL"
}
(3/15) Running awscli tests ... FAILED in 17 seconds
{
  "name": "awscli",
  "duration": 11242,
  "function": "delete_bucket\n",
  "status": "FAIL",
  "error": "delete: s3://awscli-mint-test-bucket-29546/datafile-1-kBremove_bucket failed: s3://awscli-mint-test-bucket-29546 An error occurred (BucketNotEmpty) when calling the DeleteBucket operation: The bucket you tried to delete is not empty"
}
(3/15) Running healthcheck tests ... done in 0 seconds
(4/15) Running mc tests ... done in 47 seconds
(5/15) Running minio-dotnet tests ... done in 1 minutes and 30 seconds
(6/15) Running minio-go tests ... FAILED in 3 seconds
{
  "args": {
    "bucketName": "minio-go-test-m5cscleowac92v2x",
    "objectName": "zp4ukcttbngwdvrjuq2sagteyvg1dp",
    "region": "eu-west-1"
  },
  "duration": 2778,
  "error": "The bucket you tried to delete is not empty",
  "function": "MakeBucket(bucketName, region)",
  "message": "Cleanup failed",
  "name": "minio-go: testGetObjectClosedTwiceV2",
  "status": "FAIL"
}
(6/15) Running minio-java tests ... FAILED in 1 seconds
{
  "name": "minio-java",
  "function": "listBuckets()",
  "duration": 26,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = SlowDown, message = Please reduce your request, bucketName = minio-java-test-84r22s, objectName = null, resource = /minio-java-test-84r22s, requestId = 161B7837E7065B06, hostId = bd352d75-b777-4398-9fb3-de9f95bf4ae4)\nrequest={method=PUT, url=http://minio-dev7.minio.io:32037/minio-java-test-84r22s, headers=Host: minio-dev7.minio.io:32037\nAccept-Encoding: identity\nUser-Agent: MinIO (amd64; amd64) minio-java/dev\nContent-MD5: 1B2M2Y8AsgTpgAmY7PhCfg==\nx-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\nx-amz-date: 20200624T114434Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20200624/us-east-1/s3/aws4_request, SignedHeaders=accept-encoding;content-md5;host;x-amz-content-sha256;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=503, headers=Accept-Ranges: bytes\nContent-Length: 303\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nRetry-After: 120\nServer: MinIO/DEVELOPMENT.2020-06-24T11-28-44Z\nVary: Origin\nX-Amz-Request-Id: 161B7837E7065B06\nX-Xss-Protection: 1; mode=block\nDate: Wed, 24 Jun 2020 11:44:34 GMT\n}\n >>> [io.minio.MinioClient.execute(MinioClient.java:1211), io.minio.MinioClient.execute(MinioClient.java:1236), io.minio.MinioClient.executePut(MinioClient.java:1410), io.minio.MinioClient.makeBucket(MinioClient.java:3286), io.minio.MinioClient.makeBucket(MinioClient.java:3199), FunctionalTest.listBuckets_test(FunctionalTest.java:324), FunctionalTest.runTests(FunctionalTest.java:4011), FunctionalTest.main(FunctionalTest.java:4192)]"
}
(6/15) Running minio-js tests ... FAILED in 44 seconds
{
  "name": "minio-js",
  "function": "\"after all\" hook in \"functional tests\"",
  "duration": 6,
  "status": "FAIL",
  "error": "S3Error: Please reduce your request at Object.parseError (node_modules/minio/dist/main/xml-parsers.js:86:11) at /mint/run/core/minio-js/node_modules/minio/dist/main/transformers.js:156:22 at DestroyableTransform._flush (node_modules/minio/dist/main/transformers.js:80:10) at DestroyableTransform.prefinish (node_modules/readable-stream/lib/_stream_transform.js:129:10) at prefinish (node_modules/readable-stream/lib/_stream_writable.js:611:14) at finishMaybe (node_modules/readable-stream/lib/_stream_writable.js:620:5) at endWritable (node_modules/readable-stream/lib/_stream_writable.js:643:3) at DestroyableTransform.Writable.end (node_modules/readable-stream/lib/_stream_writable.js:571:22) at IncomingMessage.onend (_stream_readable.js:673:10) at endReadableNT (_stream_readable.js:1201:12) at processTicksAndRejections (internal/process/task_queues.js:84:21)"
}
(6/15) Running minio-py tests ... FAILED in 6 minutes and 2 seconds
{
  "name": "minio-py:test_fput_object_small_file",
  "function": "fput_object(bucket_name, object_name, file_path, content_type, metadata, sse, progress, part_size)",
  "args": {
    "bucket_name": "minio-py-test-63c53880-9320-4d47-88d8-297ba0f3341c",
    "object_name": "8c685b2e-2644-4d7f-a1ab-f93fd5a0c930",
    "file_path": "/mint/data/datafile-1-MB",
    "metadata": {
      "x-amz-storage-class": "STANDARD_IA"
    }
  },
  "duration": 240389,
  "message": "BucketNotEmpty: message: The bucket you tried to delete is not empty.",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 371, in test_fput_object_small_file\n    client.remove_bucket(bucket_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 446, in remove_bucket\n    self._url_open('DELETE', bucket_name=bucket_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 2020, in _url_open\n    object_name).get_exception()\nminio.error.BucketNotEmpty: BucketNotEmpty: message: The bucket you tried to delete is not empty.\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 1983, in main\n    test_fput_object_small_file(client, testfile, log_output)\n  File \"/mint/run/core/minio-py/tests.py\", line 373, in test_fput_object_small_file\n    raise Exception(err)\nException: BucketNotEmpty: message: The bucket you tried to delete is not empty.\n",
  "status": "FAIL"
}
(6/15) Running s3cmd tests ... done in 21 seconds
(7/15) Running s3select tests ... FAILED in 40 minutes and 5 seconds
{
  "name": "s3select:test_csv_input_quote_char",
  "function": "select_object_content(bucket_name, object_name, opts)",
  "args": {
    "bucket_name": "s3select-test-9e4fe394-d75e-4f00-98c4-9d67177fe455"
  },
  "duration": 2404060,
  "message": "BucketNotEmpty: message: The bucket you tried to delete is not empty.",
  "error": "Traceback (most recent call last):\n  File \"./tests.py\", line 45, in main\n    test_csv_input_custom_quote_char(client, log_output)\n  File \"/mint/run/core/s3select/csv.py\", line 110, in test_csv_input_custom_quote_char\n    client.remove_bucket(bucket_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 446, in remove_bucket\n    self._url_open('DELETE', bucket_name=bucket_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 2020, in _url_open\n    object_name).get_exception()\nminio.error.BucketNotEmpty: BucketNotEmpty: message: The bucket you tried to delete is not empty.\n",
  "status": "FAIL"
}
(7/15) Running security tests ... done in 0 seconds

Executed 7 out of 15 tests successfully.
```
<a name="c7dbda91-89b4-46d3-8f10-200350931661"></a>
### 9902-5e62a2b/mint-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev7.minio.io:32257
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 1ec0513ba660:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 1 seconds
(2/15) Running aws-sdk-java tests ... done in 2 seconds
(3/15) Running aws-sdk-php tests ... done in 42 seconds
(4/15) Running aws-sdk-ruby tests ... done in 3 seconds
(5/15) Running awscli tests ... done in 2 minutes and 24 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 31 seconds
(8/15) Running minio-dotnet tests ... done in 1 minutes and 0 seconds
(9/15) Running minio-go tests ... FAILED in 58 seconds
{
  "args": {
    "bucketName": "minio-go-test-2omw3bdw0jwsbw0o",
    "objectName": "hjef4sql54jzetigykm0jxc4xth4fl",
    "objectPrefix": "",
    "recursive": "true"
  },
  "duration": 2404,
  "error": "The bucket you tried to delete is not empty. You must delete all versions in the bucket.",
  "function": "RemoveObjectsWithOptions(bucketName, objectsCh, opts)",
  "message": "Cleanup failed",
  "name": "minio-go: testRemoveObjectsWithOptions",
  "status": "FAIL"
}
(9/15) Running minio-java tests ... FAILED in 1 minutes and 55 seconds
{
  "name": "minio-java",
  "function": "enableObjectLegalHold(String bucketName, String objectName, String versionId)",
  "duration": 153,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = BucketNotEmpty, message = The bucket you tried to delete is not empty. You must delete all versions in the bucket., bucketName = minio-java-test-2evmoq4, objectName = null, resource = /minio-java-test-2evmoq4, requestId = 161B7815C2A59E78, hostId = f287b04d-93c2-4fd3-9e5a-66f95a3ae606)\nrequest={method=DELETE, url=http://minio-dev7.minio.io:32257/minio-java-test-2evmoq4, headers=Host: minio-dev7.minio.io:32257\nAccept-Encoding: identity\nUser-Agent: MinIO (amd64; amd64) minio-java/dev\nContent-MD5: 1B2M2Y8AsgTpgAmY7PhCfg==\nx-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\nx-amz-date: 20200624T114208Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20200624/us-east-1/s3/aws4_request, SignedHeaders=accept-encoding;content-md5;host;x-amz-content-sha256;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=409, headers=Accept-Ranges: bytes\nContent-Length: 373\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nServer: MinIO/DEVELOPMENT.2020-06-24T11-28-44Z\nVary: Origin\nX-Amz-Request-Id: 161B7815C2A59E78\nX-Xss-Protection: 1; mode=block\nDate: Wed, 24 Jun 2020 11:42:08 GMT\n}\n >>> [io.minio.MinioClient.execute(MinioClient.java:1211), io.minio.MinioClient.execute(MinioClient.java:1236), io.minio.MinioClient.executeDelete(MinioClient.java:1366), io.minio.MinioClient.removeBucket(MinioClient.java:3686), FunctionalTest.enableObjectLegalHold_test(FunctionalTest.java:3314), FunctionalTest.runTests(FunctionalTest.java:4079), FunctionalTest.main(FunctionalTest.java:4192)]"
}
(9/15) Running minio-js tests ... done in 45 seconds
(10/15) Running minio-py tests ... done in 2 minutes and 21 seconds
(11/15) Running s3cmd tests ... done in 19 seconds
(12/15) Running s3select tests ... done in 3 seconds
(13/15) Running security tests ... done in 0 seconds

Executed 13 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally

--- a/docs/sts/keycloak.md
+++ b/docs/sts/keycloak.md
@@ -57,6 +57,7 @@ Set `identity_openid` config with `config_url`, `client_id` and restart MinIO
 ```
 ~ mc admin config set myminio identity_openid config_url="http://localhost:8080/auth/realms/demo/.well-known/openid-configuration" client_id="account"
 ```
+> Note: You can configure the `scopes` parameter to restrict the OpenID scopes requested by minio to the IdP, for example, `"openid,policy_role_attribute"`, being `policy_role_attribute` a client_scope / client_mapper that maps a role attribute called policy to a `policy` claim returned by Keycloak
 
 Once successfully set restart the MinIO instance.
 ```
@@ -86,6 +87,8 @@ This will open the login page of keycloak, upon successful login, STS credential
   }
 }
 ```
+
+> Note: You can use the `-cscopes` parameter to restrict the requested scopes, for example to `"openid,policy_role_attribute"`, being `policy_role_attribute` a client_scope / client_mapper that maps a role attribute called policy to a `policy` claim returned by Keycloak.
 
 These credentials can now be used to perform MinIO API operations.
 

--- a/docs/sts/web-identity.go
+++ b/docs/sts/web-identity.go
@@ -30,6 +30,7 @@ import (
 	"log"
 	"net/http"
 	"net/url"
+	"strings"
 	"time"
 
 	"golang.org/x/oauth2"
@@ -79,6 +80,7 @@ var (
 	configEndpoint string
 	clientID       string
 	clientSec      string
+	clientScopes   string
 	port           int
 )
 
@@ -131,6 +133,7 @@ func init() {
 		"OpenID discovery document endpoint")
 	flag.StringVar(&clientID, "cid", "", "Client ID")
 	flag.StringVar(&clientSec, "csec", "", "Client Secret")
+	flag.StringVar(&clientScopes, "cscopes", "openid", "Client Scopes")
 	flag.IntVar(&port, "port", 8080, "Port")
 }
 
@@ -148,6 +151,11 @@ func main() {
 		return
 	}
 
+	scopes :=  ddoc.ScopesSupported
+	if clientScopes != "" {
+		scopes = strings.Split(clientScopes, ",");
+	}
+
 	ctx := context.Background()
 
 	config := oauth2.Config{
@@ -158,7 +166,7 @@ func main() {
 			TokenURL: ddoc.TokenEndpoint,
 		},
 		RedirectURL: fmt.Sprintf("http://localhost:%d/oauth2/callback", port),
-		Scopes:      ddoc.ScopesSupported,
+		Scopes:      scopes,
 	}
 
 	state := randomState()

--- a/docs/sts/web-identity.md
+++ b/docs/sts/web-identity.md
@@ -95,6 +95,8 @@ export MINIO_ACCESS_KEY=minio
 export MINIO_SECRET_KEY=minio123
 export MINIO_IDENTITY_OPENID_CONFIG_URL=https://accounts.google.com/.well-known/openid-configuration
 export MINIO_IDENTITY_OPENID_CLIENT_ID="843351d4-1080-11ea-aa20-271ecba3924a"
+# Optional: Allow to specify the requested OpenID scopes (OpenID only requires the `openid` scope)
+#export MINIO_IDENTITY_OPENID_SCOPES="openid,profile,email"
 minio server /mnt/export
 ```
 

